modules/log.xqm

```diff
@@ -0,0 +1,66 @@
+xquery version "3.1";
+
+
+module namespace log="http://exist-db.org/xquery/app/log";
+
+
+import module namespace config="http://exist-db.org/xquery/apps/config" at "config.xqm";
+
+
+declare function log:event($event as element(event)) as empty-sequence() {
+    let $today := current-date()
+    let $log-document := log:doc($today)
+
+    let $update-log :=
+        if (doc-available($log-document)) then 
+            update insert $event into doc($log-document)/public-repo-log
+        else ( 
+            log:mkcol($config:logs-col, log:subcollection($today)),
+            xmldb:store(log:collection($today), 
+                log:document-name($today), 
+                element public-repo-log { $event })
+        )
+
+    return
+        ()
+};
+
+declare function log:doc($date as xs:date) {
+    log:collection($date) || "/" || log:document-name($date)
+};
+
+declare function log:subcollection($date as xs:date) {
+    format-date($date, "[Y]/[M01]")
+};
+
+declare function log:collection($date as xs:date) {
+    $config:logs-col || "/" || 
+    log:subcollection($date)
+};
+
+declare function log:document-name($date as xs:date) {
+    ``[public-repo-log-`{format-date($date, "[Y]-[M01]-[D01]")}`.xml]``
```

Comment:
The original formulation in `config:log-doc-name` was more readable. Let's only resort to the string constructor to avoid escaping curly braces, etc.

Reply:
I am so used to read template strings that it is actually the easier form to read.
What about a two line form with a intermediate variable:
```xquery
let $formatted-date := format-date($date, "[Y]-[M01]-[D01]")
return
  ``[public-repo-log-`{$formatted-date}`.xml]``
```

I can revert that, but I strongly disagree that string constructors should only be used to avoid escaping curly braces. Think of multiline strings and templates in general.
For me it is the other way around, I do use concatenation (`||`) only when a string constructor does not make sense.

Reply:
Are you going to change the whole app then or just this one?

Reply:
No, I will revert it :)

Reply:
Thanks! (Phew!)